psi4/src/psi4/libmints/basisset.cc

```diff
@@ -547,7 +547,12 @@ std::string BasisSet::print_detail_cfour() const
         for (int am = 0; am <= max_am_center; am++) {
             // Write exponents for each shell
             for (size_t ep = 0; ep < exp_per_am[am].size(); ep++) {
-                sprintf(buffer, "%14.7f", exp_per_am[am][ep]);
+                if (exp_per_am[am][ep] < 100000)
+                    sprintf(buffer, "%14.7f", exp_per_am[am][ep]);
+                else if (exp_per_am[am][ep] >= 1000000)
+                    sprintf(buffer, "%12.5f ", exp_per_am[am][ep]);
+                else if (exp_per_am[am][ep] >= 100000)
+                    sprintf(buffer, "%13.6f ", exp_per_am[am][ep]);
```

Comment:
I would reorder these so you check largest to smallest and end in a `else` to make it clear that there no issues here.